lib/routes/zhihu/all-collections.ts

```diff
@@ -0,0 +1,114 @@
+import { Route, ViewType } from '@/types';
+import got from '@/utils/got';
+import { header } from './utils';
+import { parseDate } from '@/utils/parse-date';
+import cache from '@/utils/cache';
+
+export const route: Route = {
+    path: '/people/allCollections/:id',
+    categories: ['social-media'],
+    view: ViewType.Articles,
+    example: '/zhihu/people/allCollections/87-44-49-67',
+    parameters: { id: '作者 id，可在用户主页 URL 中找到' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: true,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['www.zhihu.com/people/:id'],
+            // target: 'people/allCollections/:id',
+        },
+    ],
+    name: '用户全部收藏内容',
+    maintainers: ['CohenV'],
+    handler,
+};
+
+async function handler(ctx) {
+    const id = ctx.req.param('id');
+    const apiPath = `https://api.zhihu.com/people/${id}/collections`;
+
+    const response = await got(String(apiPath), {
+        headers: {
+            Referer: `https://www.zhihu.com/people/${id}/collections`,
+        },
+    });
+
+    const collections = response.data.data;
+
+    const allCollectionItems = await Promise.all(
+        collections.map(async (collection) => {
+            const firstPageResponse = await got({
+                method: 'get',
+                url: `https://www.zhihu.com/api/v4/collections/${collection.id}/items?offset=0&limit=20`,
+                headers: {
+                    ...header,
+                    Referer: `https://www.zhihu.com/collection/${collection.id}`,
+                },
+            });
+
+            const {
+                data: items,
+                paging: { totals },
+            } = firstPageResponse.data;
+
+            if (totals > 20) {
+                const offsetList = Array.from({ length: Math.ceil(totals / 20) - 1 }, (_, index) => (index + 1) * 20);
+
+                const otherPages = await Promise.all(
+                    offsetList.map((offset) =>
+                        cache.tryGet(`https://www.zhihu.com/api/v4/collections/${collection.id}/items?offset=${offset}&limit=20`, async () => {
+                            const response = await got({
+                                method: 'get',
+                                url: `https://www.zhihu.com/api/v4/collections/${collection.id}/items?offset=${offset}&limit=20`,
+                                headers: {
+                                    ...header,
+                                    Referer: `https://www.zhihu.com/collection/${collection.id}`,
+                                },
+                            });
```

Comment:
Please use either 
https://github.com/DIYgod/RSSHub/blob/524e93324b707fac56406ee082ae8921af0a062a/lib/routes/zhihu/all-collections.ts#L36-L40 or https://github.com/DIYgod/RSSHub/blob/524e93324b707fac56406ee082ae8921af0a062a/lib/routes/zhihu/all-collections.ts#L66-L73, but [NOT BOTH](https://docs.rsshub.app/joinus/advanced/script-standard#general-guidelines) style.